Formula/swtpm.rb

```diff
@@ -0,0 +1,28 @@
+class Swtpm < Formula
+  desc "Software TPM Emulator based on libtpms"
+  homepage "https://github.com/stefanberger/swtpm"
+  url "https://github.com/stefanberger/swtpm/archive/refs/tags/v0.7.3.tar.gz"
+  sha256 "e856d1f5842fb3335164f02f2c545dd329efbc3416db20b7a327e991a4cd49c8"
+  license "BSD-3-Clause"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "gawk" => :build
+  depends_on "glib" => :build
+  depends_on "json-glib" => :build
+  depends_on "libtool" => :build
+  depends_on "socat" => :build
+  depends_on "libtasn1"
+  depends_on "libtpms"
+  depends_on "openssl@3"
+
+  def install
+    system "./autogen.sh", *std_configure_args, "--with-openssl"
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    system bin/"swtpm", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Updated test